sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -440,6 +441,16 @@ object StateStore extends Logging {
   @GuardedBy("loadedProviders")
   private val schemaValidated = new mutable.HashMap[StateStoreProviderId, Option[Throwable]]()
 
+  private val maintenanceThreadPoolLock = new Object
+
+  private val threadPoolException = new AtomicReference[Throwable](null)
```

Comment:
Can we add a comment here ? @HeartSaVioR - is this the preferred method to bubble up exceptions from the thread pool ?

Reply:
Yeah this was an approach outlined by Jungtaek which we also use in microbatch pipelining